diff -Nur a/arch/arm64/boot/dts/allwinner/sun50i-h616-orangepi-zero2.dts b/arch/arm64/boot/dts/allwinner/sun50i-h616-orangepi-zero2.dts
--- a/arch/arm64/boot/dts/allwinner/sun50i-h616-orangepi-zero2.dts	2023-07-16 03:42:38.000000000 +0800
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h616-orangepi-zero2.dts	2023-08-21 12:42:39.026231100 +0800
@@ -19,6 +19,10 @@
 	aliases {
 		ethernet0 = &emac0;
 		serial0 = &uart0;
+		led-boot = &led_boot;
+		led-failsafe = &led_boot;
+		led-running = &led_run;
+		led-upgrade = &led_run;
 	};
 
 	chosen {
@@ -28,7 +32,7 @@
 	leds {
 		compatible = "gpio-leds";
 
-		led-0 {
+		led_boot: led-0 {
 			function = LED_FUNCTION_POWER;
 			label = "orangepi:green:power";
 			color = <LED_COLOR_ID_RED>;
@@ -36,7 +40,7 @@
 			default-state = "on";
 		};
 
-		led-1 {
+		led_run: led-1 {
 			function = LED_FUNCTION_STATUS;
 			label = "orangepi:red:status";
 			color = <LED_COLOR_ID_GREEN>;
